src/test/scala/org/broadinstitute/dsde/agora/server/webservice/EntityCreationPermissionSpec.scala

```diff
@@ -0,0 +1,189 @@
+
+    assertResult(expectedNs2Perms) {ns2perms}
+  }
+
+  it should "allow creating a new entity if you have permissions on the namespace" in {
+    // owner1 create in namespace1
+    assertEntityCreation(owner1.get, namespace1.get, randUUID)
+  }
+
+  it should "reject creating a new entity if you lack permissions on the namespace" in {
+    // owner1 create in namespace2
+    assertEntityRejection(owner1.get, namespace2.get, randUUID)
+  }
+
+  it should "allow creating a new snapshot of an existing entity if you own the entity" in {
+    // owner1 create new snapshot of testEntity1WithId in namespace1
+    assertEntityCreation(owner1.get, testEntity1WithId.namespace.get, testEntity1WithId.name.get)
+  }
+
+  it should "reject creating a new snapshot of an existing entity if you do not own the entity" in {
+    // owner3 create random new entity in namespace2: prove that owner3 has Create in the namespace
+    assertEntityCreation(owner3.get, namespace2.get, randUUID)
+
+    // assert that owner3 does not have any permissions on pre-existing testEntity2WithId
+    val existingPerms = patiently(permissionBusiness.listEntityPermissions(testEntity2WithId, owner2.get))
+    assert(!existingPerms.exists(_.user == owner3.get), "owner3 should not have any permissions on testEntity2WithId")
+
+    // owner3 create new snapshot of testEntity2WithId in namespace2: should fail to create, because owner3 doesn't
+    // have permissions on previous snapshots.
+    assertEntityRejection(owner3.get, testEntity2WithId.namespace.get, testEntity2WithId.name.get)
+  }
+
+  it should "allow creating a new snapshot of an existing entity if you own a snapshot of the entity but lack permission on the namespace" in {
+    val name = randUUID
+    assertEntityCreation(owner1.get, namespace1.get, name)
+
+    val ent = AgoraEntity(namespace1, Some(name), Some(1))
+
+    // add owner2
+    patiently(permissionBusiness.insertEntityPermission(
+      ent, owner1.get, AccessControl(owner2.get, AgoraPermissions(All))
+    ))
+    // remove owner1
+    patiently(permissionBusiness.deleteEntityPermission(
+      ent, owner2.get, owner1.get
+    ))
+    val actualPermissions = patiently(permissionBusiness.listEntityPermissions(ent, owner2.get))
+    assertResult( List(AccessControl(owner2.get, AgoraPermissions(All))), "should have modified snapshot permissions correctly" ) {
+      actualPermissions
+    }
+    // at this point, the entity has one snapshot, owned by owner2, but owner2 does not have create permissions
+    // in the namespace.
+    assertEntityCreation(owner2.get, namespace1.get, name)
+  }
+
+  it should "allow creating a new snapshot of an existing entity if you own any previous snapshot of that entity" in {
+
+    val name = randUUID
+    assertEntityCreation(owner2.get, namespace2.get, name) // create snapshot 1
+    assertEntityCreation(owner2.get, namespace2.get, name) // create snapshot 2
+
+    val ent = AgoraEntity(namespace2, Some(name), Some(2))
```

Comment:
Might be nice to call this `entitySnapshot2` to more obviously indicate that `Some(2)` is specifying the second snapshot